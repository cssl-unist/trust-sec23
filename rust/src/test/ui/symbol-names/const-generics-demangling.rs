// build-fail
// compile-flags: -Z symbol-mangling-version=v0

#![feature(min_const_generics, rustc_attrs)]

pub struct Unsigned<const F: u8>;

#[rustc_symbol_name]
//~^ ERROR symbol-name(_RMCs4fqI2P2rA04_25const_generics_demanglingINtB0_8UnsignedKhb_E)
//~| ERROR demangling(<const_generics_demangling[317d481089b8c8fe]::Unsigned<11: u8>>)
//~| ERROR demangling-alt(<const_generics_demangling::Unsigned<11>>)
impl Unsigned<11> {}

pub struct Signed<const F: i16>;

#[rustc_symbol_name]
//~^ ERROR symbol-name(_RMs_Cs4fqI2P2rA04_25const_generics_demanglingINtB2_6SignedKsn98_E)
//~| ERROR demangling(<const_generics_demangling[317d481089b8c8fe]::Signed<-152: i16>>)
//~| ERROR demangling-alt(<const_generics_demangling::Signed<-152>>)
impl Signed<-152> {}

pub struct Bool<const F: bool>;

#[rustc_symbol_name]
//~^ ERROR symbol-name(_RMs0_Cs4fqI2P2rA04_25const_generics_demanglingINtB3_4BoolKb1_E)
//~| ERROR demangling(<const_generics_demangling[317d481089b8c8fe]::Bool<true: bool>>)
//~| ERROR demangling-alt(<const_generics_demangling::Bool<true>>)
impl Bool<true> {}

pub struct Char<const F: char>;

#[rustc_symbol_name]
//~^ ERROR symbol-name(_RMs1_Cs4fqI2P2rA04_25const_generics_demanglingINtB3_4CharKc2202_E)
//~| ERROR demangling(<const_generics_demangling[317d481089b8c8fe]::Char<'∂': char>>)
//~| ERROR demangling-alt(<const_generics_demangling::Char<'∂'>>)
impl Char<'∂'> {}

fn main() {}
