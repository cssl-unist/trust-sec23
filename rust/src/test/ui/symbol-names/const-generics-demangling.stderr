error: symbol-name(_RMCs4fqI2P2rA04_25const_generics_demanglingINtB0_8UnsignedKhb_E)
  --> $DIR/const-generics-demangling.rs:8:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: demangling(<const_generics_demangling[317d481089b8c8fe]::Unsigned<11: u8>>)
  --> $DIR/const-generics-demangling.rs:8:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: demangling-alt(<const_generics_demangling::Unsigned<11>>)
  --> $DIR/const-generics-demangling.rs:8:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: symbol-name(_RMs_Cs4fqI2P2rA04_25const_generics_demanglingINtB2_6SignedKsn98_E)
  --> $DIR/const-generics-demangling.rs:16:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: demangling(<const_generics_demangling[317d481089b8c8fe]::Signed<-152: i16>>)
  --> $DIR/const-generics-demangling.rs:16:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: demangling-alt(<const_generics_demangling::Signed<-152>>)
  --> $DIR/const-generics-demangling.rs:16:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: symbol-name(_RMs0_Cs4fqI2P2rA04_25const_generics_demanglingINtB3_4BoolKb1_E)
  --> $DIR/const-generics-demangling.rs:24:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: demangling(<const_generics_demangling[317d481089b8c8fe]::Bool<true: bool>>)
  --> $DIR/const-generics-demangling.rs:24:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: demangling-alt(<const_generics_demangling::Bool<true>>)
  --> $DIR/const-generics-demangling.rs:24:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: symbol-name(_RMs1_Cs4fqI2P2rA04_25const_generics_demanglingINtB3_4CharKc2202_E)
  --> $DIR/const-generics-demangling.rs:32:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: demangling(<const_generics_demangling[317d481089b8c8fe]::Char<'∂': char>>)
  --> $DIR/const-generics-demangling.rs:32:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: demangling-alt(<const_generics_demangling::Char<'∂'>>)
  --> $DIR/const-generics-demangling.rs:32:1
   |
LL | #[rustc_symbol_name]
   | ^^^^^^^^^^^^^^^^^^^^

error: aborting due to 12 previous errors

